### ISSUE TITLE
Consider underlying IRI when linking to homonyms in an ontology

### ISSUE THREAD

**Issue by mcourtot** [protegeproject/protege#585]
For example, in GO, we have 2 classes labeled 'protein'. One imported from ChEBI, one imported from PRO. When clicking on logical axioms to understand inference Protege seem to choose either of those classes.

For example, even though protein binding in GO is defined as 
id: GO:0005515
name: protein binding
intersection_of: GO:0005488 ! binding
intersection_of: has_input PR:000000001 ! protein

clicking on 'protein' in the restriction displayed in Protege leads me to the ChEBI:protein class.

cc @balhoff @cmungall @dosumis

**Comment by dosumis** [comment]
Known problem.   Think we still have more than one cell too. I've previously suggested adding an 'obo foundry uniquename' for cases like this.  This would at least fix the Protege issue.  We could fix reasoning issue by asserting equivalent locally (although this may break some release processes).  It would be much better, of course, for there to be only one term shared by ChEBI and Pro.  That seems unlikely to happen in the foreseeable future.

**Comment by mcourtot** [comment]
> Known problem. 

I knew about the PRO/ChEBI issue, not the Protege display one. A quick search on this tracker didn't yield anything of relevance, can you please add cross-refs so history is consolidated/trackable in one place?

> Think we still have more than one cell too. I've previously suggested adding an 'obo foundry uniquename' for cases like this. This would at least fix the Protege issue. 

There are several options to temporarily address the current issue (we could also update the label locally for example). Ultimately however referencing/navigating to other classes should be done by IRI rather than label. Non uniqueness of label is one of the reasons we chose to use numerical IDs (and not everybody would implement obo unique labels even if they were implemented...)

> We could fix reasoning issue by asserting equivalent locally (although this may break some release processes). It would be much better, of course, for there to be only one term shared by ChEBI and Pro. That seems unlikely to happen in the foreseeable future.

This is more of a GO issue so suggest we port this to the GO tracker; ticket is at [Parentage of glycoprotein binding](https://github.com/geneontology/go-ontology/issues/12580) (just added link to corresponding [annotation issue](https://github.com/geneontology/go-annotation/issues/1462)). FWIW, I like the idea of asserting the equivalence locally; if it breaks (too many) things we can always revert.

**Comment by balhoff** [comment]
Right, although there are Protege workarounds, like rendering by prefixed annotation property (you can set it up so that you could see `PR:protein` and `CHEBI:protein`), I think the application should know which term you have clicked on in a given expression and select the correct one, even if they are rendered the same.

**Comment by cmungall** [comment]
Hmm, I thought we filed a ticket for this yonks ago but it appears not.

Seems to be same issue this user is having: #10

**Comment by matthewhorridge** [comment]
Fixing this requires a pretty big overhaul of rendering infrastructure.  Slated for 6.0.0.
